frontend/src/metabase/querying/filters/utils/dates.ts

```diff
@@ -222,22 +246,114 @@ function getQuarterYearFilterClause(
   });
 }
 
-export function getDatePickerOperators(
-  query: Lib.Query,
-  stageIndex: number,
-  column: Lib.ColumnMetadata,
-): DatePickerOperator[] {
-  return Lib.filterableColumnOperators(column)
-    .map(operator => Lib.displayInfo(query, stageIndex, operator).shortName)
-    .filter(isDatePickerOperator);
+export function getDateFilterDisplayName(
+  value: DateFilterValue,
+  { withPrefix }: DateFilterDisplayOpts = {},
+) {
+  return match(value)
+    .with(
+      { type: "specific", operator: "=" },
+      ({ values: [date], hasTime }) => {
+        const dateText = formatDate(date, hasTime);
+        return withPrefix ? t`On ${dateText}` : dateText;
+      },
+    )
+    .with(
+      { type: "specific", operator: "<" },
+      ({ values: [date], hasTime }) => {
+        return t`Before ${formatDate(date, hasTime)}`;
+      },
+    )
+    .with(
+      { type: "specific", operator: ">" },
+      ({ values: [date], hasTime }) => {
+        return t`After ${formatDate(date, hasTime)}`;
+      },
+    )
+    .with(
+      { type: "specific", operator: "between" },
+      ({ values: [startDate, endDate], hasTime }) => {
+        return `${formatDate(startDate, hasTime)} - ${formatDate(endDate, hasTime)}`;
+      },
+    )
+    .with({ type: "relative" }, ({ value, unit, offsetValue, offsetUnit }) => {
+      if (offsetValue != null && offsetUnit != null) {
+        const prefix = Lib.describeTemporalInterval(value, unit);
+        const suffix = Lib.describeRelativeDatetime(offsetValue, offsetUnit);
+        return t`${prefix}, starting ${suffix}`;
```

Comment:
Translation context would be useful here

Reply:
Updated